engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/bigInteger/DivideNode.java

```diff
@@ -5,32 +5,34 @@
 import com.oracle.truffle.api.nodes.Node;
 import org.enso.interpreter.dsl.BuiltinMethod;
 import org.enso.interpreter.node.expression.builtin.number.utils.BigIntegerOps;
-import org.enso.interpreter.node.expression.builtin.number.utils.ToEnsoNumberNode;
 import org.enso.interpreter.runtime.error.TypeError;
 import org.enso.interpreter.runtime.number.EnsoBigInteger;
 
 @BuiltinMethod(type = "Big_Integer", name = "/", description = "Big integer division.")
 public abstract class DivideNode extends Node {
-  private @Child ToEnsoNumberNode toEnsoNumberNode = ToEnsoNumberNode.build();
-
-  abstract Object execute(EnsoBigInteger _this, Object that);
+  abstract double execute(EnsoBigInteger _this, Object that);
 
   static DivideNode build() {
     return DivideNodeGen.create();
   }
 
   @Specialization
-  Object doLong(EnsoBigInteger _this, long that) {
-    return toEnsoNumberNode.execute(BigIntegerOps.divide(_this.getValue(), that));
+  double doBigInteger(EnsoBigInteger _this, EnsoBigInteger that) {
+    return BigIntegerOps.toDouble(_this.getValue()) / BigIntegerOps.toDouble(that.getValue());
+  }
+
+  @Specialization
+  double doLong(EnsoBigInteger _this, long that) {
+    return BigIntegerOps.toDouble(_this.getValue()) / that;
   }
 
   @Specialization
-  Object doBigInteger(EnsoBigInteger _this, EnsoBigInteger that) {
-    return toEnsoNumberNode.execute(BigIntegerOps.divide(_this.getValue(), that.getValue()));
+  double doDouble(EnsoBigInteger _this, double that) {
+    return BigIntegerOps.toDouble(_this.getValue()) / that;
   }
```

Comment:
Am I correct thinking that for big enough `BigInteger` here, the result of any division here will be `+-inf`? I'm not really sure how this could be improved, but I think this may be unintuitive to the users (that dividing two non-zero integers yields infinity), maybe this should be documented somewhere?

Reply:
You are correct. I'm not sure it requires documenting, it's just a natural consequence of using double division.

Reply:
Ah right, it makes sense since there is also the separate `div` operator, didn't think about it this way.